### ISSUE TITLE
[iOS] `onPress={Vibration.vibrate()}` doesn't do anything when button is pressed; only activates when the app is opened

### ISSUE THREAD

**Issue by hb-webdev** [facebook/react-native#32352]
## Description
I am re-opening this seemingly longstanding issue for some iOS devs (according to a few Google results): `onPress={Vibration.vibrate()}` does not do anything when the button is pressed in iOS 14/15, and no errors are thrown. Oddly, `onPress={Vibration.vibrate()}` _does_ activate **once** immediately when I open the app, but never when I actually press the button. (Same exact behavior is seen with `react-native-haptic-feedback`'s library, except it emits haptic feedback instead of a vibration.)

I just spent a day and a half updating to XCode 13 and iOS 15.0.1 with the exact same results, to make sure I felt valid in submitting this issue.
I just also created yet _another_ brand new 0.66.0 React Native project with the same code as quoted below. Same results: `Vibration.vibrate()` is activated when the app opens, but absolutely nothing happens when I press the button (no logs either).
It makes no difference whether I start the app with `npx react-native run-ios --device` or through Xcode, or with `--configuration Release`.

## React Native version:
Xcode 13 | iOS 15.0.1 | **RN 0.66.0** | macOS Big Sur 11.5.1 | iPhone 12

## Steps To Reproduce

1. `npx react-native-cli init <repo name>` (Uses latest RN 0.66.0)
2. Run the code below through Xcode or `npx react-native run-ios --device` on iPhone iOS 14.8 or 15.0.1 (I have iPhone 12).
3. Open the app. For me it vibrates upon opening the app, but does not vibrate when the button is pressed.

## Expected Results
Phone vibrates when button is pressed

## Snack, code example, screenshot, or link to a repository:
```js
import React from 'react';
import {
  View,
  Button,
  Vibration
} from 'react-native';

const App = () => {

  return (
    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}} >
  <Button
    title="Buzz!"
    onPress={Vibration.vibrate()} />
    </View>
  );
};

export default App;
```
_Reference to old issue: https://github.com/facebook/react-native/issues/23404#issuecomment-934880420_

**Comment by hramos** [comment]
Thanks for reporting this! Would you mind editing your issue to make sure it makes use of the template? It should have been automatically filled out but it looks like its missing here. You can find an example of the information we need at https://github.com/facebook/react-native/issues/new?assignees=&labels=Needs%3A+Triage+%3Amag%3A&template=bug_report.md&title=

As the person who last looked into this back on iOS 13, I want to get all the necessary details to repro this on iOS 14 and iOS 15.

**Comment by hb-webdev** [comment]
@hramos I used the "Reference in new issue" GitHub feature, so it didn't auto-populate with your guys' issue template. I edited the comment.

> I want to get all the necessary details to repro this on iOS 14 and iOS 15.

I'm afraid there's nothing noteworthy I do to produce the issue: Just create a new 0.66.0 RN project and add `{onPress={Vibration.vibrate()}` to a button

Thanks for responding. I was afraid nobody would when I saw there were 1.7k issues...
